Statements inside ROS_ASSERT_MSG are not run in release mode

calum-blair-archangel opened the pull request:
When compiling in release mode (`catkin_make -DCMAKE_BUILD_TYPE=Release`),
several `ROS_ASSERT_MSG` statements are not executed. This causes camera initialisation to fail
 and the relevant messages are not displayed. This PR moves statements which set variables
outside of the `ROS_ASSERT_MSG` statements.

calumblair committed:
Fix assertions in release mode.

When compiling in release mode (catkin_make -DCMAKE_BUILD_TYPE=Release),
several ROS_ASSERT_MSG statements are not executed. This causes initialisation to fail
 and the relevant messages are not displayed. Move statements which set variables
outside of the ROS_ASSERT_MSG statements.

ghost commented:
Fixes #80

vik748 commented:
@calum-blair-archangel Thanks for taking this on.  We only merge from dev into master.  Can you please rebase this pull request to merge into dev.

calumblair committed:
Release assertions: fix whitespace

ghost commented:
@vik748 thanks, I've updated the branch and rebased

mithundiddi committed:
Update capture.cpp

- adds missing ; on line 378
- fixes warning of printing var of size_t at two places
